treeherder/config/settings.py

```diff
@@ -88,6 +88,15 @@
     # 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 ]
 
+if DEBUG:
+    # insert debug toolbar middleware just after the gzip
+    # middleware
+    gzip_index = MIDDLEWARE_CLASSES.index(
+        'django.middleware.gzip.GZipMiddleware')
+    MIDDLEWARE_CLASSES.insert(
+        gzip_index + 1,
+        'debug_toolbar.middleware.DebugToolbarMiddleware')
```

Comment:
A pattern that bedrock uses to avoid having to insert dynamically, is:
https://github.com/mozilla/bedrock/blob/1cd3149352b410c9fb6a497e6626496e54c28685/bedrock/settings/base.py#L323-L340

...up to you which is preferable.